goal_src/jak2/pc/pckernel.gc

```diff
@@ -61,20 +62,33 @@
 
 (defmethod update-discord-rpc pc-settings-jak2 ((obj pc-settings-jak2))
   "update discord rpc module"
-  (let ((info (new 'stack 'discord-info)))
-    (set! (-> info orb-count) (&-> *game-info* skill-total))
-    (set! (-> info gem-count) (&-> *game-info* gem-total))
-    (set! (-> info death-count) (&-> *game-info* total-deaths))
-    (set! (-> info status) "Playing Jak II™")
-    ;; grab the name of the level we're in
-    (if (and *level* (level-get-target-inside *level*))
-      (set! (-> info level) (symbol->string (-> (level-get-target-inside *level*) name)))
-      (set! (-> info level) "unknown"))
-    (set! (-> info cutscene?) #f) ;; TODO - cutscenes don't work yet
-    (set! (-> info time-of-day) (&-> *time-of-day-context* time))
-    (set! (-> info percent-complete) (calculate-percentage *game-info*))
-    ;; TODO - wrapping in `with-profiler` causes an error, fix it
-    (pc-discord-rpc-update info))
+  (set! *discord-info* (new 'stack 'discord-info))
+  (set! (-> *discord-info* orb-count) (-> *game-info* skill-total))
+  (set! (-> *discord-info* gem-count) (-> *game-info* gem-total))
+  (set! (-> *discord-info* death-count) (-> *game-info* total-deaths))
+  (set! (-> *discord-info* task) "unknown")
+  (set! (-> *discord-info* status) (get-active-mission-description))
+  ;; grab the name of the level we're in
+  (if *level*
+    (cond
+      ((= (level-status *level* 'title) 'active)
+        (set! (-> *discord-info* level) "title")
+        (set! (-> *discord-info* status) "On title screen")
+        )
+      (else
+        (if (level-get-target-inside *level*)
+          (set! (-> *discord-info* level) (symbol->string (-> (level-get-target-inside *level*) name)))
+          (set! (-> *discord-info* level) "unknown")
+          )
+        )
+      )
+    )
+  (set! (-> *discord-info* cutscene?) (-> *pc-settings* movie?))
+  (set! (-> *discord-info* time-of-day) (-> *time-of-day-context* time))
+  (set! (-> *discord-info* percent-complete) (calculate-percentage *game-info*))
+  (set! (-> *discord-info* focus-status) (if *target* (-> *target* focus-status) 0))
+  ;; TODO - wrapping in `with-profiler` causes an error, fix it
+  (pc-discord-rpc-update *discord-info*)
```

Comment:
Just wondering, what's the reason for putting a stack-allocated structure in a global variable? I don't think it's safe to access it outside this scope.

Reply:
The main reason I made it a global variable was because I was lazy and wanted to set the `task` field inside `get-active-mission-description`. I can just allocate it in the global heap instead (or maybe make it a `*pc-settings*` field?), it shouldn't take up too much space.